src/util/secrets.rs

```diff
@@ -178,6 +178,14 @@ impl Display for DerivationPurpose {
     }
 }
 
+/// Generates a random 32-byte array using getrandom crate.
+/// Works on all platforms including wasm32.
+pub fn generate_random_32_bytes() -> [u8; 32] {
```

Comment:
nit: seems a duplicate of `rng_32b`, you could make `rng_32b` pub, use `getrandom` if "cfg feat wasm", otherwise `OsRng`, in `Cargo.toml` do not add `getrandom` dep for non-wasm, use `rng_32b` instead of `generate_random_32_bytes`

Reply:
I missed rng_32b, since `cargo check --target wasm32-unknown-unknown` pass, we can use it directly.
The method that was causing issue was `SessionSecretRand::from_rng` which requires rand features on secp